Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,385 @@
+     * Maximum amount of time (in seconds) to wait for a message.
+     */
+    const WAIT_TIMEOUT = 10;
+
+    /**
+     * Maximum amount of time (in seconds) to set as an interval while consuming messages
+     */
+    const WAIT_PERIOD = 0.2;
+
+    /**
+     * @var \AMQPQueue
+     */
+    private $queue;
+
+    /**
+     * @var int Time in ms
+     */
+    private $dequeueingTimeMs = 0;
+
+    /**
+     * @var string
+     */
+    private $format;
+
+    /**
+     * @var PhpAmqpLibQueueManager
+     */
+    private $manager;
+
+    /**
+     * @var string Host form Message Broker
+     */
+    private $host;
+
+    /**
+     * @var string Username to connect to Message broker
+     */
+    private $username;
+
+    /**
+     * @var string $password
+     */
+    private $password;
+
+    /**
+     * @var string $port
+     */
+    private $port;
+
+    /**
+     * @var string $vhost
+     */
+    private $vhost;
+
+    /**
+     * @var array
+     */
+    private $connectionId = [];
+
+    /**
+     * @var AMQPChannel
+     */
+    private $channel;
+
+    /**
+     * PhpAmqpLibDriver constructor.
+     * @param QueueManager $manager
+     */
+    public function __construct(PhpAmqpLibQueueManager $manager)
+    {
+        parent::__construct();
+        $this->manager = $manager;
+        $this->format = self::FORMAT_JSON;
+    }
+
+    /**
+     * PhpAmqpLibDriver destruct
+     */
+    public function __destruct()
+    {
+        $this->manager->disconnect();
+    }
+
+    /**
+     * @param $host
+     * @param string $username
+     * @param string $password
+     * @param string $format
+     * @param null $port
+     * @param null $vhost
+     */
+    public function configure($host, $username, $password, $format = self::FORMAT_JSON, $port = null, $vhost = null)
+    {
+        $this->format = $format;
+    }
+
+    /**
+     * @throws \AMQPException
+     */
+    public function connect()
+    {
+        $this->manager->connect();
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function disconnect()
+    {
+        $this->manager->disconnect();
+    }
+
+    /**
+     * @return QueueMessage|QueueMessageInterface
+     */
+    public function createQueueMessage($queueName = null, $options = [])
+    {
+        $msg = new QueueMessage();
+        $msg->setContext(new Context());
+        $this->setChannel();
+        $this->manager->declareQueue($queueName, AMQP_DURABLE, $options);
+        return $msg;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isConnected()
+    {
+        return $this->manager->isConnected();
+    }
+
+    /**
+     * @return bool
+     */
+    public function isSubscribed()
+    {
+        return null !== $this->queue;
+    }
+
+    /**
+     * @param string $queue
+     * @throws \AMQPChannelException
+     * @throws \AMQPConnectionException
+     * @throws \AMQPQueueException
+     */
+    public function subscribe($queue)
+    {
+        if ($this->currentEnvelope) {
```

Comment:
You are using this property ```currentEnvelope``` but is not defined as property in the class

Reply:
Yes, I have to implement the subscribe method. This class is here just because of the abstract parent class.